requests/models.py

```diff
@@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
         content_type = None
         length = None
 
-        if json is not None:
+        if not data and json is not None:
```

Comment:
Since we apparently can't use `if data is None` can we compare to an empty dictionary? Explicit is so much better than implicit.